### PR TITLE
Fix: Enable Slate's default variables to be used by gulp-slate's app.scss file

### PR DESCRIPTION
**Pull request by alexgoldstone**
Apologies. I didn't realise that you were referencing Slate's default variables in gulp-slate's app.scss file and my last pull request broke things.

Specifically, app.scss references $phone-width.

This pull request make's Slate's default variables (or variables the user specifies instead) available to app.scss.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -109,6 +109,7 @@ function buildAssets (opts, callback) {
             }
             rawScss.push('@import "'+opts.variables+'";');
         }
+        rawScss.push('@import "'+getModulePath('slate')+'/source/stylesheets/_variables.scss";');
```
Comment (wodka) [inline index.js:112]: I think this should be before the if condition -> you want to be able to overwrite the variables

Comment (alexgoldstone) [inline index.js:112]: No. From the SASS documentation:

> You can assign to variables if they aren’t already assigned by adding the !default flag to the end of the value. This means that if the variable has already been assigned to, it won’t be re-assigned, but if it doesn’t have a value yet, it will be given one.

So, if you specify a variable it will be used and if you don't specify a variable then the Slate variable will be used. I have tested with the order in the pull request (manually changing the Slate variables to !default) and it works.

Comment (wodka) [inline index.js:112]: yes, forgot that they include the variable.scss again from their scss files

File: index.js [hunk 2]
```diff
@@ -109,22 +109,25 @@ function buildAssets (opts, callback) {
...
         rawScss.push('@function font-url($url){ @return url($url) }');
-        rawScss.push('@media screen { @import "'+getModulePath('slate')+'/source/stylesheets/screen.css.scss"; }');
-        rawScss.push('@media screen { .highlight._{ @import "'+getModulePath('highlight.js')+'/../styles/solarized-light"; } }');
```
Comment (wodka) [inline index.js:114]: why did you remove this?

File: index.js [hunk 3]
```diff
@@ -205,7 +208,7 @@ module.exports = function (opts) {
         filename: false,
         style: 'solarized-dark',
         template: ROOT+'src/layout.html',
-        scss: ROOT+'src/app.scss',
```
Comment (wodka) [inline index.js:208]: this file contains several fixes to the slate scss

File: index.js (hunk 2 above)
Comment (alexgoldstone) [inline index.js:114]: The Slate scss is still being pulled in (see line 122 below) but only when a user hasn't specified their own scss file.
The highlight.js css is still being pulled in (see line 126 below). It wasn't working being imported and had to be referenced separately.

File: index.js (hunk 3 above)
Comment (alexgoldstone) [inline index.js:208]: My intention was to move where it was to move it not remove it - I have just done another commit to this pull request which brings it inline with how variables are used but reversed it as your CSS changes are breaking a design that otherwise works so the CSS itself needs investigation.

File: index.js (hunk 2 above)
Comment (wodka) [inline index.js:114]: but you only import one, the second one is a forced style (just wasn't yet configureable) to have code blocks in the center area .highlight._{}

File: index.js (hunk 3 above)
Comment (wodka) [inline index.js:208]: you mean the current version does not work in one case? can you explain further?

**Comment by wodka** [comment]
I changed now the handling of the original slate files to apply our own variables without the modifications on the original slate.

thanks for helping :)
